### PR TITLE
Adjust validation to account for new udsverfc and udsverlc rule

### LINKED ISSUE #721: C2 UDSVERFC or UDSVERLC validation bug
We have validation on the C2 so that if `UDSVERFC` or `UDSVERLC` are 95-98 then `UDSVERTN`,` UDSVERTE`, and `UDSVERTI` other variables should be empty. This validation has been implemented, but there is required field validation that is causing a bug.

`UDSVERFC` or `UDSVERLC` are 95-98
Image

Validation is correctly indicating that the three fields should be blank
Image

However, there is still required field validation so it is not possible for the form to be finalized
Image

- [ ] Ensure we have Playwright tests covering this section of the C2 for all test cases (not just this bug fix).

### PR DESCRIPTION
**Pull request by mlan225**
Resolves #721 

### Add new validation to the C2 form
Josh discovered new validation added to the C2 form that needed to be added to our existing validation logic

image

- [ ] Add new validation logic to C2 form (both in person and remote views for the C2)
- [ ] Add new validation messages to C2 view
- [ ] Add new validation messages to C2T view
- [ ] Helper text added under labels for questions: UDSVERTN, UDSVERTE, UDSVERTI

### Comments
- The label helper text for the in person and remote views references their corresponding views
- The error text refers to the property name due to the properties being different question numbers on each view
- The validation for UDSVERFC = 95 - 98 triggers on finalize, while UDSVERLC already disables the fields on 95 - 98

### CODE REVIEW

File: src/UDS.Net.Forms/Models/UDS4/C2.cs [hunk 1]
```diff
@@ -1132,6 +1129,24 @@ public bool? OTrailBValidation
...
+            {
+                if ((UDSVERFC.HasValue && UDSVERFC >= 95) || (UDSVERLC.HasValue && UDSVERLC.Value >= 95))
+                {
+                    if (UDSVERTN == null && UDSVERTE == null && UDSVERTI == null)
+                    {
+                        return true;
+                    }
+                }
+
+                return null;
```
Comment (mlan225) [inline src/UDS.Net.Forms/Models/UDS4/C2.cs:1200]: current client side validation already covers disabling the fields if UDSVERLC is 95 - 98, however, keeping the validation logic for both UDSVERFC and UDSVERLC in this method

File: src/UDS.Net.Forms/Pages/UDS4/_C2T.cshtml [hunk 2]
```diff
@@ -1230,35 +1230,50 @@
...
-                        <submission-error location="UDSVERTN" errors="Model.BaseForm.UnresolvedErrors" />
-                        <span class="mt-2 text-sm text-red-600" asp-validation-for="C2.UDSVERTNValidation"></span>
-                        <submission-error location="UDSVERTNValidation" errors="Model.BaseForm.UnresolvedErrors" />
+                        <div>
+                            <span class="mt-2 text-sm text-red-600" asp-validation-for="C2.UDSVERTN"></span>
+                            <span class="mt-2 text-sm text-red-600" asp-validation-for="C2.UDSVERFCReasonCodeValidation"></span>
+                            <submission-error location="UDSVERTN" errors="Model.BaseForm.UnresolvedErrors" />
+                            <span class="mt-2 text-sm text-red-600" asp-validation-for="C2.UDSVERTNValidation"></span>
+                            <submission-error location="UDSVERTNValidation" errors="Model.BaseForm.UnresolvedErrors" />
+                        </div>
```
Comment (mlan225) [inline src/UDS.Net.Forms/Pages/UDS4/_C2T.cshtml:1250]: added spans and submission error tags to a div so they properly align on the right column like in the _C2 view

**Comment by mlan225** [comment]
### Self Review:
- [x] Review / comment on code 
- [x] Test that validation now appears when providing a value for UDSVERTN, UDSVERTE, and UDSVERTI when UDSVERFC or UDSVERLC are 95 - 98
- [x] Test that validation does NOT trigger when leaving UDSVERTN, UDSVERTE, and UDSVERTI blank when UDSVERFC or UDSVERLC are 95 - 98
- [x] Test that new validation works within both the in person and remote views (C2 and C2T)
- [x] Verify that the data persists in the verbal fluency section with new validation applied

**Review by ashleybot — CHANGES_REQUESTED** [review]
@mlan225 Needs Playwright tests to ensure the correct scenarios for this section can be set and the form successfully finalize.

**Comment by mlan225** [comment]
### Tests Created

- [ ] Using both UDSVERFC and UDSVERLC error codes should disable all fields following them within section
- [ ] Using valid values for UDSVERFC and UDSVERLC should enable all fields following them within section
- [ ] If UDSVERFC (correct f-words) is between 95 and 98 or UDSVERLC (I-words correct) is between 95 and 98, then UDSVERTN, UDSVERTE, and UDSVERTI must be blank
- [ ] If UDSVERFC not 95-98 and UDSVERLC not 95-98, UDSVERTN must be the total of UDSVERFC and UDSVERLC
- [ ] When UDSVERFC and UDSVERLC are reason codes, the form saves with null values for the rest of the section
- [ ] If UDSVERNF and UDSVERLN are both within valid ranges (0-15) then UDSVERTI should equal the sum of UDSVERNF and UDSVERLN.
- [ ] If UDSVERFN (f-words repeated) is between 0 and 15 and UDSVERLR (l-words repeated) is between 0 and 15, then UDSVERTE must be the total of UDSVERFN and UDSVERLR
- [ ] When UDSVERFC and UDSVERLC are values, All questions can be given a value and save the form

**Review by ashleybot — CHANGES_REQUESTED** [review]
@mlan225 RequiredIfValidation has been removed for `UDSVERTN`, `UDSVERTE`, and `UDSVERTI`. NotMapped validation properties need to be added back so that the form requires those fields in the correct scenario on finalization.

I was able to finalize this incorrect scenario:

Image

**Comment by mlan225** [comment]
### Added missing validation

Added validation unmapped props for UDSVERTN and UDSVERTE. UDSVERTI already has validation if the value is required so there is no individual NULL value check needed
> If UDSVERNF and UDSVERLN are both within valid ranges (0-15) then UDSVERTI should equal the sum of UDSVERNF and UDSVERLN.

Unmapped prop `UDSVERTIValidation` makes sure that when UDSVERTI is required, it has a value

image

### Add test method for missing validation
- when UDSVERTN and UDSVERTE are required, expect error message for missing value
- existing test method `UDSVERTIRequiresSumOfUDSVERNFAndUDSVERLNWhenBothAreValues` covers UDSVERTI being null when required

**Comment by mlan225** [comment]
### Recent updates
- [x] Add validation message back to individual questions for UDSVERTN, UDSVERTE, and UDSVERTI
  - The validation message was removed earlier in development, but is now added back to displaying on the property and not just the validation summary
  - image
 
- [x] Walk through validation with Josh
  - We walked through and plugged in different production data into the section to test real scenarios
  - Also plugged in our own custom answers to test edge cases (this is how the missing range validation message was caught)

**Review by ashleybot — APPROVED** [review]
@mlan225 Good to bump to stable version and merge to main!
